hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClientContext.java

```diff
@@ -0,0 +1,68 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hadoop.fs.azurebfs.services;
+
+/**
+ * Class to hold extra configurations for AbfsClient and further classes
+ * inside AbfsClient.
+ */
+public class AbfsClientContext {
```

Comment:
So, just to be clear. 
- A separate builder class that would have the build method and also the with options(withAbfsCounters(), withAbfsPerfTracker(), etc), right? So, move the with options from the context class to the builder class.
- Also, the context class would now have a parameterized constructor which I would call in the build() method to return the context instance.
- The validations are for future usages, right? Since for options, I am adding I already have validation in the class that uses them( for eg: if(abfsCounter !=null) {} is checked in the AbfsClient class).

Reply:
ok, don't worry about validation. What you have done is great -all that is left is to make the AbfsClientContext package private so forcing everything to go through the builder